### ISSUE TITLE
ADD does not understand ENV variables

### ISSUE THREAD

**Issue by cybersiddhu** [GoogleContainerTools/kaniko#691]
Hi,
Kaniko build is failing when i am trying add(ADD) a remote file using ENV variable. Here is a test case.

```dockerfile
FROM busybox
ENV TEST_PLAN https://raw.githubusercontent.com/GoogleContainerTools/kaniko/master/docs/testplan.md
RUN echo $TEST_PLAN
ADD $TEST_PLAN ./plan.md
```
Here is the error 
```
error building image: error building stage: lstat /workspace/https:/raw.githubusercontent.com/GoogleContainerTools/kaniko/master/docs/testplan.md: no such file or directory
```
I am using local dir `workspace` as my build context.

The above `Dockerfile` builds fine with both `docker` and `buildah`.
Am i missing something here or kaniko does not work with this instruction.

**Comment by tejal29** [comment]
@cybersiddhu Thanks for trying kaniko.

This is related to #718 where `IsSrcsValid ` assumes source destination is in the build context.

 `IsSrcsValid ` should if src is a remote file url. if resolvedSources is a remote file then it should skip this check and fetch the file.

https://github.com/GoogleContainerTools/kaniko/blob/c8fabdf6e43b19f6a223f1d0b06e127d0774bd7e/pkg/util/command_util.go#L227

Feel free to submit a PR and we will take a look.

**Comment by N4rm0** [comment]
Hi @tejal29,

I submitted a quick PR that should fix the issue.

By activating kaniko debug log on the OP's example, here is what I found out:

```         
INFO[0006] ADD $TEST_PLAN ./plan.md                     
DEBU[0006] Resolved $TEST_PLAN to https:/raw.githubusercontent.com/GoogleContainerTools/kaniko/master/docs/testplan.md 
DEBU[0006] Resolved ./plan.md to plan.md
```
I had to squint my eyes to see that the protocol in the url is missing a slash after environment replacement.

The incriminating function is:

https://github.com/GoogleContainerTools/kaniko/blob/d67a8225031f2809f0d6c8c4f1825f233d8574a1/pkg/util/command_util.go#L65

I made a small fix, but IHMO we could have cleaner code by refactoring/removing `isFilepath` flag.

I can amend this PR in that sense.
